File: server/src/schemas/market.py
from datetime import datetime
from typing import List
from pydantic import BaseModel

class MarketItem(BaseModel):
    name: str
    current_price: float
    change: float
    change_percent: float

class MarketOverviewResponse(BaseModel):
    indices: List[MarketItem]
    risk_indicators: List[MarketItem]
    bonds: List[MarketItem]
    fx: List[MarketItem]
    commodities: List[MarketItem]
    generated_at: datetime

File: client/src/types/market.ts
export type MarketItem = {
  name: string;
  current_price: number;
  change: number;
  change_percent: number;
};

export type MarketOverviewResponse = {
  indices: MarketItem[];
  risk_indicators: MarketItem[];
  bonds: MarketItem[];
  fx: MarketItem[];
  commodities: MarketItem[];
  generated_at: string;
};
